gravitee-am-cypress/cypress/integration/oidc/ciba/ciba.spec.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+    createAcceptAllDeviceNotifier,
+    createDomain,
+    createUser,
+    requestAdminAccessToken,
+    requestDomain, startDomain
+} from "../../utils/management-utils";
+import {domainOidcCibaConfig, oidcApplication} from "../../../fixtures/oidc";
+import {User} from "../../../model/users";
+
+const configureDomainForCiba = (domain, accessToken, method, body) =>
+    requestDomain(domain, accessToken, "PATCH", body);
+
+const createCibaApplication = (domainHrid) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/register`,
+    method: 'POST',
+    retryOnStatusCodeFailure: true,
+    body: oidcApplication
+});
+
+const registerDomainWithNotifier = (domain, domainHrid, clientId, clientSecret) => cy.request({
+    url: Cypress.env().cibaDelegateNotifierUrl + "/domains",
+    method: 'POST',
+    body: {
+        domainId: domain,
+        domainCallback: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate/callback`,
+        clientId: clientId,
+        clientSecret: clientSecret
+    }
+})
+
+const requestAuthentication = (domainHrid, clientId, clientSecret, loginHint, userCode) => cy.request({
+    url: Cypress.env().gatewayUrl + `/${domainHrid}/oidc/ciba/authenticate`,
+    method: 'POST',
+    auth: {
+        user: clientId,
+        password: clientSecret
+    },
+    form: true,
+    body: {
+        scope: "openid",
+        login_hint: loginHint,
+        user_code: userCode
+    },
+    failOnStatusCode: false
+})
+
+describe("An application is correctly configured to use CIBA", () => {
+    let accessToken;
+    let domain;
+    let domainHrid;
+    let clientId;
+    let clientSecret;
+    let auth_req_id;
+    let cibaConfig;
+    const user: User = {
+        firstName: "Pat",
+        lastName: "Test",
+        email: "pat@test.com",
+        username: "pat@test.com",
+        password: "Password123",
+        additionalInformation: {
+            user_code: "1234"
+        }
+    };
+
+    before(() => {
+        requestAdminAccessToken().should(response => expect(response.status).to.eq(200))
```

Comment:
:thinking: is it possible to have something more fluent with cypress ?
I'm afraid that this "multi level" calls become hard to maintain.

Reply:
To avoid this kind of nesting you could also do something like this:
```
requestAdminAccessToken()
      .then( response => (accessToken = response.body.access_token))
      .then(() => createDomain(accessToken, "cibatest", "A test domain for CIBA"))
      .then((response) => { 
[...]
```

Reply:
That's much better.